NitroxPatcher/Patches/Dynamic/BeaconLabel_SetLabel_Patch.cs

```diff
@@ -0,0 +1,25 @@
+using System.Reflection;
+using NitroxClient.GameLogic;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+/// <summary>
+/// Adds a callback to broadcast beacon label change when edited.
+/// </summary>
+public sealed partial class BeaconLabel_OnHandClick_Patch : NitroxPatch, IDynamicPatch
+{
+    private static readonly MethodInfo TARGET_METHOD = Reflect.Method((BeaconLabel t) => t.OnHandClick(default));
+
+    public static void Postfix(BeaconLabel __instance)
+    {
+        uGUI.main.userInput.callback += _ =>
```

Comment:
Is the callback ever cleared or are we just adding callback after callback?

Reply:
The callback is initially a new instance created during the patched function onto which we add our own callback. When ``uGUI.main.userInput.RequestString`` happens (the code called in the patched function), all the input fields (callback included) are replaced by the new values provided. So everytime this function happens, it's always a new callback